### DISCUSSION TITLE
help needed : perfomance issue on OSX M1/M2 in FOR loops

### DISCUSSION THREAD

**Discussion by alaingdl** [gnudatalanguage/gdl#1684] [General]
As mentioned is #1635  and #1498 we do have a very serious performance problem on OSX with M1 or M2 processors with Clang compiler. The problem don't exist with Linux based M2 or with Clang on x86. (I was not able to compile with gcc on M2 :(

A summary : 
Normal good computer : 
```
GDL> tic & for i=1, 32700 do a=i & toc
% Time elapsed : 0.0037379265 seconds.
```

Bad M1/M2 OSX Clang : 
```
GDL> tic & for i=1, 32700 do a=i & toc
% Time elapsed : 0.19751906 seconds.
```

**Comment by GillesDuvert** [comment]
Quoting @alaingdl "(I was not able to compile with gcc on M2 :(" I had the same experience, and it gave the impression that was because all the needed libraries are compile with clang (Apple one or other, I do not know) and thus are incompatible.

**Comment by alaingdl** [comment]
In fact all kinds of loops are impacted :) 

M2 : 

```
GDL> i=100000L & tic & repeat i-- until i LT 0 & toc
% Time elapsed : 1.1988881 seconds.
GDL> i=100000L & tic & WHILE (i GT 0) DO i-- & toc
% Time elapsed : 1.1963608 seconds.
GDL> tic & for i=0, 100000L do a=i & toc
% Time elapsed : 0.60171604 seconds.
```

x86
```
GDL>  i=100000L & tic & repeat i-- until i LT 0 & toc
% Time elapsed : 0.014533997 seconds.
GDL> i=100000L & tic & WHILE (i GT 0) DO i-- & toc
% Time elapsed : 0.013131142 seconds.
GDL> tic & for i=0, 100000L do a=i & toc
% Time elapsed : 0.0040760040 seconds.


```

**Reply by GillesDuvert** [reply]
very interesting...

**Reply by alaingdl** [reply]
@GillesDuvert 
I consider I have reached my level of incompetence on that topic, I have just no idea how to debug or solve that :(
(a colleague deterred another problem with large arrays allocation in Fortran on M1/M2 ...)

I just hope to find an opportunity to recompile everything with GCC on the same computer, but I have intermittent access ...

@pjb7687 @slayoo 
Let's deliver ASAP a 1.0.4 with all the great progress we do have already, but communicate on the limits !

**Reply by pjb7687** [reply]
@alaingdl @slayoo @GillesDuvert I've just released 1.0.4. Thanks a lot for the comments!

**Comment by alaingdl** [comment]
Thanks ! I just send a message to the users ! Later I will send a message on Debian Astro list ;)
Who in in touch with the packagers ?!
